### PR TITLE
Update knative to Release 0.8

### PR DESCRIPTION
**Pull request by johnugeorge**
With the new release, all awaiting fixes from knative/pkg will be available.   

Related: 
https://github.com/kubeflow/kfserving/issues/276 and https://github.com/kubeflow/kfserving/issues/249

@jinchihe The openapi-codegen changes are not complete(added for successful build).  Once this PR is merged, you could make the necessary changes.  what do you think?

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/282)

### CODE REVIEW

**Comment by ellis-bigelow** [comment]
Great work Johnu. I imagine there will be a followon CL deprecating the use of the alpha packages "knservingv1alpha1"?

**Comment by johnugeorge** [comment]
@ellis-bigelow  knative still defaults to v1alpha1 in the release 0.8 - https://github.com/knative/serving/releases/download/v0.8.0/serving.yaml

File: pkg/apis/serving/v1alpha1/openapi_generated.go [hunk 1]
```diff
@@ -23,53 +23,53 @@ limitations under the License.
 package v1alpha1
 
 import (
-	openapispec "github.com/go-openapi/spec"
+	spec "github.com/go-openapi/spec"
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha1/openapi_generated.go]: any reason to change this ? spec is the default alias name though.

File: hack/update-openapigen.sh [hunk 2]
```diff
@@ -24,11 +24,11 @@ fi
 
 # Update kative files to add `+k8s:openapi-gen=true` to workaround the knative types cannot be load.
 # TBD @jinchihe: This should be updated from knative (PR: https://github.com/knative/pkg/pull/510).
-sed -i '/^type Condition struct {/i // +k8s:openapi-gen=true' vendor/github.com/knative/pkg/apis/condition_types.go
-sed -i '/^type VolatileTime struct {/i // +k8s:openapi-gen=true' vendor/github.com/knative/pkg/apis/volatile_time.go
+sed -i '/^type Condition struct {/i // +k8s:openapi-gen=true' vendor/knative.dev/pkg/apis/condition_types.go
```
[the remark this answers is not in the document]
Comment (johnugeorge) [reply hack/update-openapigen.sh]: I think so. However, my openapi-gen changes are only to have successful build. Once this is merged, @jinchihe can make specific changes for openapi-gen

File: pkg/apis/serving/v1alpha1/openapi_generated.go (hunk 1 above)
Comment (johnugeorge) [reply pkg/apis/serving/v1alpha1/openapi_generated.go]: Same reason as above. @jinchihe will make specific changes needed for openapi-gen

**Comment by jinchihe** [comment]
> @jinchihe The openapi-codegen changes are not complete(added for successful build). Once this PR is merged, you could make the necessary changes. what do you think?

@johnugeorge Sure, thanks!

File: hack/update-openapigen.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jinchihe) [reply hack/update-openapigen.sh]: @yuzisun Yes, absolutely right! Will remove while updating openapi and python sdk once the PR merged, thanks!

File: pkg/apis/serving/v1alpha1/openapi_generated.go (hunk 1 above)
Comment (jinchihe) [reply pkg/apis/serving/v1alpha1/openapi_generated.go]: @johnugeorge If you updated the openapi_generated.go by running the script `hack/update-openapigen.sh`, should no changes, seems this is updated [here](https://github.com/kubeflow/kfserving/blob/fb4a31ddf1bce6a0c2c1b6c65068981bb93b309b/hack/update-openapigen.sh#L34). Personally you can leave the file, just upgrade knative, I can do this later. Or of you want to do that, I can work this with you together. Thanks.

Comment (johnugeorge) [reply pkg/apis/serving/v1alpha1/openapi_generated.go]: updated

**Comment by yuzisun** [comment]
Great work @johnugeorge!  Looking forward to upgrading kfserving controller to use knative v1beta1 API.
/lgtm
/approve

**Pull request merged**